src/App.tsx

```diff
@@ -14,11 +17,144 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  dataServer: string[],
+  visibility: boolean;
+  isReverse: boolean;
+  sortType: string;
+  sizeFild: number;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    dataServer: [...goodsFromServer],
+    visibleGoods: [...goodsFromServer],
+    visibility: false,
+    isReverse: false,
+    sortType: 'none',
+    sizeFild: goodsFromServer.length,
+  };
+
+  reset = () => {
+    this.setState(state => ({
+      isReverse: false,
+      sortType: 'none',
+      sizeFild: state.dataServer.length,
+    }));
+  };
+
+  render() {
+    const {
+      dataServer,
+      visibility,
+      isReverse,
+      sortType,
+      sizeFild,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__wrap">
+          <h1 className="App__title">Goods</h1>
+          <p className="App__subtitle">
+            Length list:
+            {' '}
+            {sizeFild}
+          </p>
+          <br />
+
+          {!visibility && (
+            <button
+              className="App__btn"
+              type="button"
+              onClick={() => {
+                this.setState(state => ({
+                  visibility: !state.visibility,
+                }));
+              }}
+            >
+              Start
+            </button>
+          )}
+
+          {visibility && (
+            <div className="App__control">
+              <div className="App__list-wrap" style={{ height: `${10 * 38}px` }}>
```

Comment:
it is better to avoid using inline styles. better move it to `App__list-wrapp` class